### PR TITLE
Port distilbert transformer checkpoint

### PR DESCRIPTION
**Pull request by cosmo3769**
Hi @mattdangerw @ariG23498,

Ported distilbert transformers checkpoint in kerasNLP. Please check. Thank you!

### CODE REVIEW

File: keras_nlp/src/utils/transformers/convert_distilbert.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    )
+
+    return backbone
+
+
+def load_distilbert_backbone(cls, preset, load_weights):
+    transformers_config = load_config(preset, HF_CONFIG_FILE)
+    keras_config = convert_backbone_config(transformers_config)
+    backbone = cls(**keras_config)
+    print(backbone.summary())
```
Comment (mattdangerw) [inline keras_nlp/src/utils/transformers/convert_distilbert.py:163]: remove prints!

File: keras_nlp/src/utils/transformers/convert_distilbert.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    if load_weights:
+        jax_memory_cleanup(backbone)
+        with SafetensorLoader(preset) as loader:
+            convert_weights(backbone, loader)
+    return backbone
+
+
+def load_distilbert_tokenizer(cls, preset):
+    return cls(
+        get_file(preset, "vocab.txt"),
```
Comment (mattdangerw) [inline keras_nlp/src/utils/transformers/convert_distilbert.py:174]: should we handle `do_lower_case` like we do for bert?

File: keras_nlp/src/utils/transformers/convert_distilbert.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+from keras_nlp.src.utils.preset_utils import load_config
+from keras_nlp.src.utils.transformers.safetensor_utils import SafetensorLoader
+
+
+def convert_backbone_config(transformers_config):
+    return {
+        "vocabulary_size": transformers_config["vocab_size"],
+        "num_layers": transformers_config["n_layers"],
+        "num_heads": transformers_config["n_heads"],
+        "hidden_dim": transformers_config["hidden_dim"] // 4,
```
Comment (mattdangerw) [inline keras_nlp/src/utils/transformers/convert_distilbert.py:28]: shouldn't this just be `transformers_config["dim"]`?

File: keras_nlp/src/utils/transformers/convert_distilbert.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        "intermediate_dim": transformers_config["dim"] * 4,
```
Comment (mattdangerw) [inline keras_nlp/src/utils/transformers/convert_distilbert.py:29]: shouldn't this just be `transformers_config["hidden_dim"]`?

**Review by mattdangerw — COMMENTED** [review]
looks good overall! couple comments.

File: keras_nlp/src/utils/transformers/convert_distilbert.py (hunk 2 above)
Comment (cosmo3769) [reply keras_nlp/src/utils/transformers/convert_distilbert.py:174]: Yeah. Fixed.

File: keras_nlp/src/utils/transformers/convert_distilbert.py (hunk 3 above)
Comment (cosmo3769) [reply keras_nlp/src/utils/transformers/convert_distilbert.py]: Right. Fixed.

File: keras_nlp/src/utils/transformers/convert_distilbert.py (hunk 4 above)
Comment (cosmo3769) [reply keras_nlp/src/utils/transformers/convert_distilbert.py]: Right. Fixed.

**Review by mattdangerw — APPROVED** [review]
LGTM! Running a quick end to end test, which merge if all looks good.

**Pull request merged**